state/firewallrules.go

```diff
@@ -86,59 +110,66 @@ func NewFirewallRules(st *State) *firewallRulesState {
 
 // Save stores the specified firewall rule.
 func (fw *firewallRulesState) Save(rule FirewallRule) error {
-	if err := rule.WellKnownService.validate(); err != nil {
+	if err := checkModelActive(fw.st); err != nil {
 		return errors.Trace(err)
 	}
-	for _, cidr := range rule.WhitelistCIDRs {
+	buildTxn := func(int) ([]txn.Op, error) {
+		return fw.GetTransactions(rule, false)
+	}
+	if err := fw.st.db().Run(buildTxn); err != nil {
+		return errors.Annotate(err, "failed to create firewall rules")
+	}
+
+	return nil
+}
+
+func (fw *firewallRulesState) GetTransactions(rule FirewallRule, migrationWanted bool) ([]txn.Op, error) {
```

Comment:
This is not returning multiple transactions. It is returning operations to run in a transaction. A better name is something like `GetSaveTransactionOps`, which also indicates the purpose of said ops.

Shorten the argument name to`isMigrating` and indicate its purpose in a comment for this exported method.